compiler/fir/semantics/src/org/jetbrains/kotlin/fir/resolve/dfa/model.kt

```diff
@@ -12,56 +12,18 @@ import kotlin.contracts.contract
 
 // --------------------------------------- Facts ---------------------------------------
 
-operator fun TypeStatement.plus(other: TypeStatement?): TypeStatement = other?.let { this + other } ?: this
-
 class MutableTypeStatement(
     override val variable: RealVariable,
```

Comment:
I agree that `exactNotType` is not used anywhere, but I originally introduced for possible applications in the future (maybe it will be further improvements of DFA, maybe it will be useful for some static checkers, IDK)

And I think that it's easier to incrementally support this field instead of introducing it in all places if/when we decide to really use it (and it's not present in the code)

So me, personally, against this commit

Reply:
@semoro, @mglukhikh WDYT about this particular change?

Reply:
I think it'll be easier to reintroduce `exactNotType` after some cleanup, because ideally it really shouldn't have any effect other than a bunch of new implications.

Currently the biggest use of it is the `returns() implies` checker, which is broken because of that, as disjunctions on types are by necessity approximating. As a result it produced incorrect warnings even on simple statements with negated disjunctions. The approximation needs to be carefully thought out to avoid such errors.

Reply:
Ok, aggree